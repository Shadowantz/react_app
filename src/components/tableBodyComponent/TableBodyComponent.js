import React from 'react';
import * as R from 'ramda';
import { connect } from 'react-redux';

import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import TableBody from '@material-ui/core/TableBody';

function generateTableRow(elem) {
	const { ticketType, ticketNumber, ticketSummary, assignee, reporter, priority, status, estimate } = elem;

	return (
		<TableRow
			tabIndex={-1}
			key={ticketNumber}
		>
			<TableCell
				id={ticketNumber}
				size="small"
			>
				{ticketType}
			</TableCell>
			<TableCell align="center" size="small">{ticketNumber}</TableCell>
			<TableCell size="small">{ticketSummary}</TableCell>
			<TableCell size="small">{assignee}</TableCell>
			<TableCell size="small">{reporter}</TableCell>
			<TableCell size="small">{priority}</TableCell>
			<TableCell size="small">{status}</TableCell>
			<TableCell align="center" size="small">{estimate}</TableCell>
		</TableRow>
	);
}

export function TableBodyComponent(props) {
	const { displayElements } = props;

	if (!displayElements) return null;

	return (
		<TableBody>
			{R.map((tableElement) => generateTableRow(tableElement), displayElements)}
		</TableBody>
	);
}

const TableBodyComponentConnected = connect(
	(state) => {
		const { displayElements } = state;

		return {
			displayElements,
		};
	},
)(TableBodyComponent);

export default TableBodyComponentConnected;
