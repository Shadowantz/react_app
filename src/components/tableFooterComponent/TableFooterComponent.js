import React, { useState } from 'react';
// import * as R from 'ramda';
import { connect } from 'react-redux';

import TableRow from '@material-ui/core/TableRow';
import TableCell from '@material-ui/core/TableCell';
import TablePagination from '@material-ui/core/TablePagination';
import { TableFooter } from '@material-ui/core';
import InputBase from '@material-ui/core/InputBase';
import IconButton from '@material-ui/core/IconButton';
import SearchIcon from '@material-ui/icons/Search';

import { setPage, setRowsPerView, filterBySearch } from '../../reducers';
import { debounceUtility } from '../../utils/utils.js';

export function TableFooterComponent(props) {
	const { elementsCount, page, rowsPerView, setRowsPerViewAction, setPageAction, filterBySearchAction } = props;
	const [searchText, setSearchText] = useState('');

	const filterOnEnter = (keyCode) => {
		if (keyCode === 'Enter') {
			filterBySearchAction(searchText);
		}
	};

	const filterOnKeyPress = (value) => {
		if (!value) {
			filterBySearchAction('');
		}

		setSearchText(value);
	};

	return (
		<TableFooter>
			<TableRow>
				<TableCell
					colSpan={3}
				>
					<InputBase
						className="inputClass"
						placeholder="Search..."
						onChange={(event) => debounceUtility(filterOnKeyPress(event.target.value), 300)}
						onKeyPress={(event) => debounceUtility(filterOnEnter(event.key), 300)}
						type="search"
					/>
					<IconButton
						aria-label="search"
						onClick={() => filterBySearchAction(searchText)}
					>
						<SearchIcon />
					</IconButton>
				</TableCell>
				<TablePagination
					colSpan={6}
					rowsPerPageOptions={[5, 10]}
					count={elementsCount}
					rowsPerPage={rowsPerView}
					page={page}
					backIconButtonProps={{
						'aria-label': 'previous page',
					}}
					nextIconButtonProps={{
						'aria-label': 'next page',
					}}
					onChangePage={(event, newPage) => setPageAction(newPage)}
					onChangeRowsPerPage={(event) => setRowsPerViewAction(event.target.value)}
				/>
			</TableRow>
		</TableFooter>
	);
}

const TableFooterComponentConnected = connect(
	(state) => {
		const { page, rowsPerView, tableElements } = state;

		return {
			page,
			rowsPerView,
			elementsCount: tableElements.length,
		};
	},
	{
		setPageAction: setPage,
		setRowsPerViewAction: setRowsPerView,
		filterBySearchAction: filterBySearch,
	},
)(TableFooterComponent);

export default TableFooterComponentConnected;
