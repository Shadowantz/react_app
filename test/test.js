import React from 'react';
import { TableHeadComponent } from '../src/components/tableHeadComponent/TableHeadComponent';
import { TableBodyComponent } from '../src/components/tableBodyComponent/TableBodyComponent';
import { TableFooterComponent } from '../src/components/tableFooterComponent/TableFooterComponent';
import renderer from 'react-test-renderer';

test('Expect to render the table head component', () => {
	const component = renderer.create(
		<TableHeadComponent />,
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Expect to render the table body component', () => {
	const component = renderer.create(
		<TableBodyComponent />,
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Expect to render the table footer component', () => {
	const component = renderer.create(
		<TableFooterComponent
			props={{
				rowsPerPageOptions: [5, 10],
				count: 10,
				rowsPerPage: 5,
				page: 0
			}}
		/>,
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
