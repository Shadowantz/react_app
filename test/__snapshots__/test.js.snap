// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render the table body component 1`] = `null`;

exports[`Expect to render the table footer component 1`] = `
<tfoot
  className="MuiTableFooter-root"
>
  <tr
    className="MuiTableRow-root MuiTableRow-footer"
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-footer"
      colSpan={3}
    >
      <div
        className="MuiInputBase-root inputClass"
        onClick={[Function]}
        onKeyPress={[Function]}
      >
        <input
          className="MuiInputBase-input MuiInputBase-inputTypeSearch"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Search..."
          type="search"
        />
      </div>
      <button
        aria-label="search"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </span>
      </button>
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTablePagination-root MuiTableCell-footer"
      colSpan={6}
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
      >
        <div
          className="MuiTablePagination-spacer"
        />
        <span
          className="MuiTypography-root MuiTablePagination-caption MuiTypography-caption MuiTypography-colorInherit"
        >
          Rows per page:
        </span>
        <div
          className="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            className="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input MuiInputBase-inputSelect"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            onAnimationStart={[Function]}
            type="hidden"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <span
          className="MuiTypography-root MuiTablePagination-caption MuiTypography-caption MuiTypography-colorInherit"
        >
          NaN-NaN of undefined
        </span>
        <div
          className="MuiTablePagination-actions"
        >
          <button
            aria-label="previous page"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="next page"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </td>
  </tr>
</tfoot>
`;

exports[`Expect to render the table head component 1`] = `
<thead
  className="MuiTableHead-root"
>
  <tr
    className="MuiTableRow-root MuiTableRow-head"
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-sizeSmall MuiTableCell-head"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Ticket Type
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-sizeSmall MuiTableCell-head"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Ticket Number
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-sizeSmall MuiTableCell-head"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Ticket Summary
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-sizeSmall MuiTableCell-head"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Assignee
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-sizeSmall MuiTableCell-head"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Reporter
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-sizeSmall MuiTableCell-head"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Priority
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-sizeSmall MuiTableCell-head"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Status
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-sizeSmall MuiTableCell-head"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Estimate
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
  </tr>
</thead>
`;
